### PR TITLE
`ERC20SessionKeyValidator` Module

### PR DESCRIPTION
**Pull request by lbw33**
## Description

- `ERC20SessionKeyValidator` created that allows users to create Session Keys that specify:
       - Specific ERC20 token
       - Specific ERC165 interface id
       - Specific function selector
       - Spending limit per transaction
       - Creation and Expiry
       - Paused flag
- `ERC20SessionKeyValidator` has functionality to do the following:
       - Enabled/Disabled
       - Rotate/Pause
       - Validation of UserOps
- Test suite and some test utils.
- Added to deployment script.

## Motivation and Context

-

## How Has This Been Tested?

- Locally using test suite.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: DEPLOYMENTS_MODULAR.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+| `ModularEtherspotWalletFactory` | `sepolia` | [0x77e4288a4b15893f520f15c262a07df9866904e4](https://sepolia.etherscan.io/address/0x77e4288a4b15893f520f15c262a07df9866904e4) | [0x0f4b4f53a2654e34e7b0a3aad01f9a90d711df2af1d3d9d8a2e70a63c36bc254](https://sepolia.etherscan.io/tx/0x0f4b4f53a2654e34e7b0a3aad01f9a90d711df2af1d3d9d8a2e70a63c36bc254) |
+| `Bootstrap` | `sepolia` | [0x4f695ad7694863c8280fcebf2cb220e361ce4ea0](https://sepolia.etherscan.io/address/0x4f695ad7694863c8280fcebf2cb220e361ce4ea0) | [0xc69aed3fd25ef9bd66c42903dd855cd363d703f03049be9fb3a2fdf0c75ecfb1](https://sepolia.etherscan.io/tx/0xc69aed3fd25ef9bd66c42903dd855cd363d703f03049be9fb3a2fdf0c75ecfb1) |
+| `MultipleOwnerECDSAValidator` | `sepolia` | [0x1e714c551fe6234b6ee406899ec3be9234ad2124](https://sepolia.etherscan.io/address/0x1e714c551fe6234b6ee406899ec3be9234ad2124) | [0x1b56d1e32cb12d16b3878eed660ae23eb454cd3e84ae9e88d8ed40d0d0c5e6f8](https://sepolia.etherscan.io/tx/0x1b56d1e32cb12d16b3878eed660ae23eb454cd3e84ae9e88d8ed40d0d0c5e6f8) |
```
Comment (ch4r10t33r) [inline DEPLOYMENTS_MODULAR.md:19]: Can we pls update README to demonstrate how SessionKeys can be used? Also, do we have benchmarks on the gas associated with session keys deployment.

File: src/modular-etherspot-wallet/modules/validator/MultipleOwnerECDSAValidator.sol [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
 import {ECDSA} from "solady/src/utils/ECDSA.sol";
 import {EIP712} from "solady/src/utils/EIP712.sol";
 
 contract MultipleOwnerECDSAValidator is EIP712, IValidator {
     using ExecutionLib for bytes;
     using ECDSA for bytes32;
 
-    bytes32 internal constant EIP712_DOMAIN_TYPEHASH = 0x8b73c3c69bb8fe3d512ecc4cf759cc79239f7b179b0ffacaa9a75d522b39400f;
+    bytes32 internal constant EIP712_DOMAIN_TYPEHASH =
+        0x8b73c3c69bb8fe3d512ecc4cf759cc79239f7b179b0ffacaa9a75d522b39400f;
```
Comment (ch4r10t33r) [inline src/modular-etherspot-wallet/modules/validator/MultipleOwnerECDSAValidator.sol:17]: What is the domain hash being used here? Please can we add that as a comment.

Comment (lbw33) [reply src/modular-etherspot-wallet/modules/validator/MultipleOwnerECDSAValidator.sol]: done

File: DEPLOYMENTS_MODULAR.md (hunk 1 above)
Comment (lbw33) [reply DEPLOYMENTS_MODULAR.md:19]: done

**Review by ch4r10t33r — APPROVED** [review]
LGTM

**Review by 0xSulpiride — APPROVED** [review]
LGTM.
I'm still not sure about adding EIP-712 in `validateUserOp` because eip-712 is used by other contracts to validate signatures of the owner of a smart contract wallet. I think we only have to expose `isValidSignature` for that.

**Pull request merged**